### PR TITLE
feat: Calculate max goroutines based on file limit

### PR DESCRIPTION
**Pull request by roneli**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: helpers/limit/limits.go [hunk 1]
```diff
@@ -26,14 +32,48 @@ func GetMaxGoRoutines() uint64 {
...
 
+// DiagnoseLimits verifies if user should increase ulimit or max file descriptors to improve number of expected
+// goroutines in CQ to improve performance
+func DiagnoseLimits() (diags diag.Diagnostics) {
+	// the amount of goroutines we want based on machine memory
+	want := calculateGoRoutines(getMemory())
+	// calculate file descriptor limit
+	fds, err := calculateFileLimit()
+	if err == nil && fds < want {
+		diags = diags.Add(diag.NewBaseError(errors.New("available file descriptors capacity lower than expected"),
```
Comment (disq) [inline helpers/limit/limits.go:43]: since it's a "consider" this should be `diag.WARNING`? Same for the one below too

Comment (roneli) [reply helpers/limit/limits.go:43]: correct! my bad, even the one below should be warning

Comment (roneli) [reply helpers/limit/limits.go:43]: fixed

File: helpers/limit/limits.go [hunk 2]
```diff
@@ -26,14 +32,50 @@ func GetMaxGoRoutines() uint64 {
...
+
+func calculateGoRoutines(totalMemory uint64) uint64 {
+	total := calculateMemoryGoRoutines(totalMemory)
+	mfo, err := calculateFileLimit()
+	if err != nil {
+		return total
+	}
+
+	if mfo < total {
+		return uint64(float64(mfo) * 0.3)
```
Comment (bbernays) [inline helpers/limit/limits.go:78]: Should the `0.3` also be a `const`?

**Review by disq — APPROVED** [review]

**Review by bbernays — APPROVED** [review]

**Pull request merged**